front/lib/api/assistant/messages.ts

```diff
@@ -734,3 +735,51 @@ export async function fetchMessageInConversation(
     ],
   });
 }
+
+export async function softDeleteUserMessage(
+  auth: Authenticator,
+  {
+    messageId,
+    conversation,
+  }: {
+    messageId: string;
+    conversation: ConversationWithoutContentType;
+  }
+): Promise<Result<{ success: true }, ConversationError>> {
+  const user = auth.getNonNullableUser();
+  const owner = auth.getNonNullableWorkspace();
+
+  const message = await fetchMessageInConversation(
+    auth,
+    conversation,
+    messageId
+  );
+
+  if (!message || !message.userMessage) {
+    return new Err(new ConversationError("message_not_found"));
+  }
+
+  if (message.userMessage.userId !== user.id) {
+    return new Err(new ConversationError("message_deletion_not_authorized"));
```

Comment:
admins can also delete message (just mentioning depends if you want to do in another PR or not)